pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +27,277 @@ import (
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a cohort.
+	Spec StorageCohortSpec `json:"spec,omitempty"`
+
+	// Most recently observed status of the cohort.
+	// Populated by the cohort operator or cohort manager.
+	Status StorageCohortStatus `json:"status,omitempty"`
+}
+
+// StorageCohortSpec describes the attributes that a cohort is created with.
+type StorageCohortSpec struct {
+	// NodeSelector is used to specify the cohort to be considered
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+
+	// CohortManager stores the details of the cohort manager responsible for
+	// managing the cohort
+	CohortManager CohortManagerDetails `json:"cohortManager,omitempty"`
+
+	// StorageProvisioner contains list of all provisioners responsible for
+	// the provisioning tasks for different storage solutions in the cohort
+	StorageProvisioner []StorageCohortStorageProvisionerDetails `json:"storageProvisioner,omitempty"`
+}
+
+// StorageCohortStorageProvisionerDetails stores the different storage provisioners information
+// which takes the job pf provisioning pools and volumes
+type StorageCohortStorageProvisionerDetails struct {
+	// StorageType represents the type of storage solution which the provisioner is responsible for
+	// implementing
+	// For example: lvm or device-localpv
+	StorageType string `json:"storageType,omitempty"`
+
+	// provisioner is the driver expected to handle this cohort.
+	// This is an optionally-prefixed name, like a label key.
+	// For example: "openebs.io/scp-lvm-provisioner" or "openebs.io/scp-device-provisioner".
+	Provisioner string `json:"provisioner,omitempty"`
+}
+
+// CohortManagerDetails is information about the cohort manager managing
+// the cohort.
+type CohortManagerDetails struct {
+	// ApiUrl is the cohort manager endpoint used for communicating with the cohort
+	ApiUrl string `json:"apiUrl,omitempty"`
+
+	// HealthCheckUrl describes a health check endpoint of cohort manager against which
+	// an api call is to be performed to determine whether it is alive or ready to receive traffic.
+	HealthCheckUrl string `json:"healthCheckUrl,omitempty"`
+	// TODO add authentication & security spec
+}
+
+// StorageCohortStatus is information about the current status of a storage cohort.
+type StorageCohortStatus struct {
+	// StorageCohortPhase is the recently observed lifecycle phase of the cohort.
+	StorageCohortPhase StorageCohortPhase `json:"phase,omitempty"`
+
+	// Conditions is an array of current observed cohort component's conditions.
+	Conditions []ComponentCondition `json:"conditions,omitempty"`
+
+	// Capabilities represent capabilities that a cohort consists of
+	Capabilities StorageCohortCapabilities `json:"capabilities,omitempty"`
+}
+
+// StorageCohortCapabilities lists a set of capabilities present within the cohort.
+type StorageCohortCapabilities struct {
```

Comment:
Consider moving this to a separate re-usable structure as decided.

Reply:
Will do once the common structure is ready.